crates/wascc-provider/src/states/crash_loop_backoff.rs

```diff
@@ -1,20 +1,30 @@
-use kubelet::state::{State, Transition};
-use kubelet::{
-    pod::{Phase, Pod},
-    state,
-};
-
-use crate::{make_status, PodState};
+use crate::PodState;
+use kubelet::state::prelude::*;
 
 use super::registered::Registered;
 
-state!(
-    /// Pod has failed multiple times.
-    CrashLoopBackoff,
-    PodState,
-    {
+/// Pod has failed multiple times.
+#[derive(Default, Debug)]
+pub struct CrashLoopBackoff;
+
+#[async_trait::async_trait]
+impl State<PodState> for CrashLoopBackoff {
+    async fn next(
+        &self,
+        _pod_state: &mut PodState,
+        _pod: &Pod,
+    ) -> anyhow::Result<Transition<PodState>> {
         tokio::time::delay_for(std::time::Duration::from_secs(60)).await;
-        Ok(Transition::Advance(Box::new(Registered)))
-    },
-    { make_status(Phase::Pending, "CrashLoopBackoff") }
-);
+        Ok(Transition::next(self, Registered))
+    }
+
+    async fn json_status(
+        &self,
+        _pod_state: &mut PodState,
+        _pod: &Pod,
+    ) -> anyhow::Result<serde_json::Value> {
+        make_status(Phase::Pending, "CrashLoopBackoff")
+    }
+}
+
+impl EdgeTo<Registered> for CrashLoopBackoff {}
```

Comment:
Could the `EdgeTo` trait be better exposed with a `derive` macro in the future?

Reply:
I’m definitely open to exploring other ways of setting the edge trait, I’ve never actually written a derive macro so I’m not sure.